guide-app/src/components/screens/HomeScreen/index.js

```diff
@@ -89,107 +103,100 @@ class HomeScreen extends Component<Props> {
+              <NavigationListItem
+                index={index}
+                item={item}
+                onPressItem={this.onPressItem}
+              />
+            );
+          }}
+          keyExtractor={item => item.id}
+          data={items}
+        />
         <TouchableOpacity
-          style={styles.sectionFooterButton}
-          onPress={() => this.onPressViewAll(section.category)}
+          style={styles.mapButton}
+          onPress={() => navigation.navigate("CategoryMapScreen")}
         >
-          <Text style={styles.sectionFooterText}>
-            {LangService.strings.VIEW_ALL.toUpperCase()}
-          </Text>
+          <Image style={styles.mapIcon} source={mapIcon} />
         </TouchableOpacity>
-      </View>
-    );
-  };
-
-  render() {
-    if (this.props.showLoadingSpinner) {
-      return <ActivityIndicator style={styles.loadingSpinner} />;
-    }
-
-    // TODO move to mapStateToProps()
-    const { navigationSections, noContent } = this.props;
-    const sections = [];
-    navigationSections.forEach(cat => {
-      const data = cat.items
-        .filter(item => item.guide || item.guideGroup)
-        .sort(compareDistance)
-        .slice(0, 2);
-      if (data.length > 0) {
-        sections.push({ title: cat.name, data, category: cat });
-      } else if (noContent) {
-        sections.push({
-          title: cat.name,
-          data,
-          category: cat,
-          showSpinner: true,
-          hideFooter: true
-        });
-      }
-    });
-
-    return (
-      <SectionList
-        style={styles.container}
-        stickySectionHeadersEnabled={false}
-        renderSectionHeader={({ section }) => this.renderSectionHeader(section)}
-        renderItem={({ item }) => (
-          <NavigationListItem item={item} onPressItem={this.onPressItem} />
-        )}
-        renderSectionFooter={({ section }) =>
-          // $FlowFixMe flow doesn't understand me
-          this.renderSectionFooter(section)
-        }
-        keyExtractor={item => item.id}
-        sections={sections}
-      />
+      </>
     );
   }
 }
 
 function mapStateToProps(state: RootState) {
-  const { navigation, guides, guideGroups } = state;
+  const {
+    navigation,
+    guides,
+    guideGroups,
+    uiState: { currentHomeTab }
+  } = state;
   const { isFetching, navigationCategories } = navigation;
 
   const noContent = guides.items.length === 0 || guideGroups.items.length === 0;
 
+  const categories = navigationCategories.map(cat => {
+    const data = cat.items
+      .filter(item => item.guide || item.guideGroup)
+      .sort(compareDistance);
+    if (data.length > 0) {
+      return {
+        title: cat.name,
+        data,
+        category: cat
+      };
+    } else if (noContent) {
+      return {
+        title: cat.name,
+        data,
+        category: cat,
+        showSpinner: true,
+        hideFooter: true
+      };
+    }
+  });
+
+  const items = categories[currentHomeTab].data;
```

Comment:
This needs a default I believe for `currentHomeTab` when state is new?

Reply:
currentHomeTab's default is `0` in the reducer. Do you believe there is a case when the state hasn't been populated, but the HomeScreen is rendered?

Reply:
It keeps happening, at least once while switching languages and sometimes at initial startup so yeah somehow.